### PR TITLE
Handle IPs in Asset, VirtualServer, Cluster

### PR DESCRIPTION
**Pull request by ar4s**

### CODE REVIEW

File: src/ralph/ralph2_sync/subscribers.py [hunk 1]
```diff
@@ -105,6 +106,56 @@ def _get_configuration_path_from_venture_role(venture_role_id):
...
+    else:
+        del dca.management_ip
+
+
+def _handle_ips(obj, ips):
+    if ips is None:
+        return
+    management_ip = list(filter(lambda ip: ip['is_management'], ips))
+    if len(management_ip) > 1:
+        logger.info('More than one management IP ({}) for {}.'.format(
```
Comment (mkurek) [inline src/ralph/ralph2_sync/subscribers.py]: info -> error

File: src/ralph/ralph2_sync/subscribers.py [hunk 2]
```diff
@@ -105,6 +106,56 @@ def _get_configuration_path_from_venture_role(venture_role_id):
...
+        ip, created = IPAddress.objects.get_or_create(
+            address=ip_dict['address'],
+            defaults=dict(
+                hostname=ip_dict['hostname'],
+                dhcp_expose=ip_dict['dhcp_expose'],
+                is_management=False
+            )
+        )
+        if created:
+            ip.ethernet = Ethernet.objects.create(
```
Comment (mkurek) [inline src/ralph/ralph2_sync/subscribers.py]: create -> get_or_create

File: src/ralph/ralph2_sync/subscribers.py [hunk 3]
```diff
@@ -105,6 +106,56 @@ def _get_configuration_path_from_venture_role(venture_role_id):
...
+                base_object=obj, mac=ip_dict['mac']
+            )
+        else:
+            ip.ethernet.base_object = obj
+            ip.ethernet.mac = ip_dict['mac']
+            ip.ethernet.save()
+        ip.save()
```
Comment (mkurek) [inline src/ralph/ralph2_sync/subscribers.py:164]: i'd add here sth like 

```
ip.is_management = False
ip.dhcp_expose = ip_dict['dhcp_expose']
```

to make sure that everything is updated properly (when ip is not created)

File: src/ralph/ralph2_sync/subscribers.py [hunk 4]
```diff
@@ -105,6 +106,58 @@ def _get_configuration_path_from_venture_role(venture_role_id):
     return None
 
 
+def _handle_management_ip(dca, management_ip, management_hostname):
+    if management_ip:
+        print(management_ip, management_hostname)
```
Comment (szok) [inline src/ralph/ralph2_sync/subscribers.py]: print :)

**Comment by szok** [comment]
:ok:

File: src/ralph/ralph2_sync/subscribers.py [hunk 5]
```diff
@@ -105,6 +106,58 @@ def _get_configuration_path_from_venture_role(venture_role_id):
...
+            address=ip_dict['address'],
+            defaults=dict(
+                hostname=ip_dict['hostname'],
+                dhcp_expose=ip_dict['dhcp_expose'],
+                is_management=False
+            )
+        )
+        if created:
+            ip.ethernet, _ = Ethernet.objects.get_or_create(
+                base_object=obj, mac=ip_dict['mac']
```
Comment (mkurek) [inline src/ralph/ralph2_sync/subscribers.py]: if `mac` is empty and there is more than one IP without mac assigned to this obj, there will be `MultipleObjectsReturned` raised.

**Comment by mkurek** [comment]
:ok:

**Pull request merged**